spyderlib/widgets/externalshell/baseshell.py

```diff
@@ -291,7 +291,6 @@ def get_stderr(self):
     
     def write_output(self):
         self.shell.write(self.get_stdout(), flush=True)
-        QApplication.processEvents()

```

Comment:
Instead of erasing this line, could you comment it like this?

    # Commenting the line below fixes Issue 2251
    # QApplication.processEvents()

If other problems appear because of this change, it'd be easier to come back to this place later :-)